akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/scaladsl/GrpcReadJournal.scala

```diff
@@ -303,13 +314,18 @@ final class GrpcReadJournal private (
               log.debug2("{}: Filter updated [{}]", streamId, criteria.mkString(", "))
             StreamIn(StreamIn.Message.Filter(FilterReq(protoCriteria)))
 
-          case r @ ConsumerFilter.ReplayWithFilter(`streamId`, _) =>
+          case r @ ConsumerFilter.ReplayWithFilter(`streamId`, _, Some(`replayCorrelationId`)) =>
```

Comment:
in this way we ignore replay request with wrong correlation id, and hence not sent to the producer side at all